pkg/controller/atlasdatafederation/datafederation_controller.go

```diff
@@ -110,6 +109,35 @@ func (r *AtlasDataFederationReconciler) Reconcile(contextInt context.Context, re
 	}
 	ctx.Client = atlasClient
 
+	owner, err := customresource.IsOwner(dataFederation, r.ObjectDeletionProtection, customresource.IsResourceManagedByOperator, managedByAtlas(contextInt, atlasClient))
```

Comment:
My 2 cents, we could try to increase our confidence in code with more unit tests. For this particular case this piece of code could be on a separate method that we could try to test in isolation. For an example check out[ PR #1036 protection tests](https://github.com/mongodb/mongodb-atlas-kubernetes/pull/1036/files#diff-0e68b721c4dc0a2ce82467624c0d6842d6f91467b6c6bc95e1dc414e13f6c1bf)

Not a blocker anyway, just a suggestion

Reply:
I did see the unit tests you added when I was looking at your PR last week, and I'm definitely inspired. Will look to add some today!